src/components/Todos.tsx

```diff
@@ -0,0 +1,237 @@
+import React, { useEffect, useState } from "react";
+import "bootstrap/dist/css/bootstrap.css";
+import "./Todos.css";
+import { useNavigate } from "react-router-dom";
+import { ArrowUp, ArrowDown } from "lucide-react";
+import useFetch from "./useFetch";
+
+export interface Todo {
+  id: number;
+  title: string;
+  date: string;
+  isCompleted: boolean;
+}
+
+export default function Todos() {
+  let { todos, loading, error,refetchData } = useFetch("http://localhost:8000/todos");
+  const [todolist, setTodolist] = useState<Todo[]>([]);
+  const [search, setSearch] = useState<string>("");
+  const [status, setStatus] = useState("all");
+  const [sortBy, setSortBy] = useState<string>("name");
+  const [sortOrder, setSortOrder] = useState<string>("asc");
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (todos) {
+      setTodolist(todos);
+      // setNextId(todos.length + 1);
+    }
+  }, [todos]);
+
+  const handleCheckbox = (
+    todo: Todo,
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    fetch(`http://localhost:8000/todos/${todo.id}`, {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        isCompleted: e.target.checked,
+      }),
+    })
+      .then((res) => res.json())
+      .then(() => {
+          refetchData(true)
+      });
+  };
+
+  const handleDelete = (id: number) => {
+    fetch(`http://localhost:8000/todos/${id}`, {
+      method: "DELETE",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })
+      .then(() => alert("Deleted Successfully"))
+      .then(() =>    refetchData(true));
+  };
+
+  const handleStatus = (e:React.MouseEvent<HTMLButtonElement,MouseEvent>) => {
+   setStatus(e.currentTarget.value)
+  };
+
+  const filteredData = todolist.filter((item: Todo) => {
+    if (status === 'all') {
+      return true;
+    }
+    return status === 'complete' ? item.isCompleted : !item.isCompleted;
+  });
+  
+
+  const handleSort = (criteria: string) => {
+    if (criteria === sortBy) {
+      setSortOrder(sortOrder === "asc" ? "desc" : "asc");
+    } else {
+      setSortBy(criteria);
+      setSortOrder("asc");
+    }
+  };
+
+  const navigateTodo =(url:string,id:number)=>{
+    navigate(url+id)
+  }
+
+  filteredData.sort((a: Todo, b: Todo) => {
+    if (sortBy === "name") {
+      return sortOrder === "asc" ? a.title.localeCompare(b.title) : b.title.localeCompare(a.title);
+    } else if (sortBy === "date") {
+      return sortOrder === "asc" ? new Date(a.date).getTime() - new Date(b.date).getTime() : new Date(b.date).getTime() - new Date(a.date).getTime();
+    }
+    return 0;
+  });
+
+
+  const searchFilter = (item:Todo, search:string) => {
+    return search.trim() === "" ? true : item.title.toLowerCase().includes(search.toLowerCase());
```

Comment:
Filter the data for search and render directly

Reply:
Done